### ISSUE TITLE
Unable to run images with --userns-remap enabled

### ISSUE THREAD

**Issue by ziermmar** [docker/docker#31898]
**Description**
When running docker with --userns-remap I can't run any images. 

**Error message**
ERRO[0010]` Create container failed with error: oci runtime error: container_linux.go:247: starting container process caused "process_linux.go:359: container init caused \"rootfs_linux.go:54: mounting \\\"proc\\\" to rootfs \\\"/var/virtual/docker/10000.10000/devicemapper/mnt/e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233/rootfs\\\" at \\\"/proc\\\" caused \\\"operation not permitted\\\"\""

I am able to run images when i start them privileged with '--privileged --userns  host' or when I am not running with --userns-remap at all

**Debug log of docker startup**
```
root@matrix ~ # dockerd -l debug --userns-remap default --storage-driver=devicemapper --storage-opt=dm.thinpooldev=/dev/mapper/storage-docker-tpool
INFO[0000] User namespaces: ID ranges will be mapped to subuid/subgid ranges of: dockremap:dockremap
DEBU[0000] Creating user namespaced daemon root: /var/virtual/docker/10000.10000
DEBU[0000] docker group found. gid: 999
DEBU[0000] Listener created for HTTP on unix (/var/run/docker.sock)
INFO[0000] libcontainerd: new containerd process, pid: 23761
DEBU[0000] libcontainerd: containerd health check returned error: rpc error: code = 14 desc = grpc: the connection is unavailable
INFO[0001] User namespaces: ID ranges will be mapped to subuid/subgid ranges of: dockremap:dockremap
DEBU[0001] Using default logging driver json-file
DEBU[0001] Golang's threads limit set to 113400
DEBU[0001] [graphdriver] trying provided driver: devicemapper
DEBU[0001] devicemapper: driver version is 4.27.0
DEBU[0001] devmapper: Generated prefix: docker-253:12-2108032
DEBU[0001] devmapper: Checking for existence of the pool storage-docker-tpool
DEBU[0001] devmapper: loadDeviceFilesOnStart()
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/2f221bc44eb744ac7d2552f339eb0f9f64c837a863cdfa45b0fec0e078596f4a
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/2f221bc44eb744ac7d2552f339eb0f9f64c837a863cdfa45b0fec0e078596f4a-init
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/492a3547c6572d6b64f6c177dd666d978c71442ed643c3d1b8b59d80f6211663
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/492a3547c6572d6b64f6c177dd666d978c71442ed643c3d1b8b59d80f6211663-init
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/5b243a4d5c56dde124afec797c61d90d456d2777e32fe5220b359fff7ccf8289
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/5b243a4d5c56dde124afec797c61d90d456d2777e32fe5220b359fff7ccf8289-init
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/755163ca07d5b8d5fc7d626795870489e0601ebc3af5e1c7414e1725270c0ad1
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/755163ca07d5b8d5fc7d626795870489e0601ebc3af5e1c7414e1725270c0ad1-init
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/75527f523061d8458764321651c6d956a915f316ad8576e4a77616ca3060710f
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/75527f523061d8458764321651c6d956a915f316ad8576e4a77616ca3060710f-init
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/7e7a6199339b301ceb30ab998734b0aa7f39858df51d5ecdf3ccb31f8e93421a
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/7e7a6199339b301ceb30ab998734b0aa7f39858df51d5ecdf3ccb31f8e93421a-init
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/base
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/c2eca2c3d1d01bb05528edc5269f9603c91c8427222fea9d5b60371fee62c13b
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/c33c079233d33155cd6d85c786e33165ae269e634d22740ec1bb44da60f910f5
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/c33c079233d33155cd6d85c786e33165ae269e634d22740ec1bb44da60f910f5-init
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/cb8dda47ed710b3d15956a98d280dbc3803ada1ebfee3158a110b3e6f582dac6
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/cb8dda47ed710b3d15956a98d280dbc3803ada1ebfee3158a110b3e6f582dac6-init
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/d701015f6f7e86128de6782690842f47434a57038ea260938946e7865c815927
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/d701015f6f7e86128de6782690842f47434a57038ea260938946e7865c815927-init
DEBU[0001] devmapper: Skipping file /var/virtual/docker/10000.10000/devicemapper/metadata/deviceset-metadata
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/dfdf13b33fba865256c54533fd890a2a157ffe1fca28a633dd9af19e82097974
DEBU[0001] devmapper: Loading data for file /var/virtual/docker/10000.10000/devicemapper/metadata/dfdf13b33fba865256c54533fd890a2a157ffe1fca28a633dd9af19e82097974-init
DEBU[0001] devmapper: Skipping file /var/virtual/docker/10000.10000/devicemapper/metadata/transaction-metadata
DEBU[0001] devmapper: loadDeviceFilesOnStart() END
DEBU[0001] devmapper: constructDeviceIDMap()
DEBU[0001] devmapper: Added deviceId=17 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=20 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=19 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=12 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=11 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=13 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=3 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=16 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=1 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=22 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=14 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=4 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=18 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=2 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=6 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=15 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=8 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=7 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=21 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=5 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=10 to DeviceIdMap
DEBU[0001] devmapper: Added deviceId=9 to DeviceIdMap
DEBU[0001] devmapper: constructDeviceIDMap() END
DEBU[0001] devmapper: activateDeviceIfNeeded()
DEBU[0001] devmapper: UUID for device: /dev/mapper/docker-253:12-2108032-base is:3b957fb9-dbff-4f2f-9a55-864297ef7fb1
WARN[0001] devmapper: Base device already exists and has filesystem ext4 on it. User specified filesystem  will be ignored.
DEBU[0001] devmapper: deactivateDevice START()
DEBU[0001] devmapper: removeDevice START(docker-253:12-2108032-base)
DEBU[0001] devmapper: removeDevice END(docker-253:12-2108032-base)
DEBU[0001] devmapper: deactivateDevice END()
DEBU[0001] Using graph driver devicemapper
DEBU[0001] Max Concurrent Downloads: 3
DEBU[0001] Max Concurrent Uploads: 5
INFO[0001] Graph migration to content-addressability took 0.00 seconds
WARN[0001] Your kernel does not support kernel memory limit
WARN[0001] Your kernel does not support cgroup cfs period
WARN[0001] Your kernel does not support cgroup cfs quotas
WARN[0001] Your kernel does not support cgroup rt period
WARN[0001] Your kernel does not support cgroup rt runtime
WARN[0001] mountpoint for pids not found
INFO[0001] Loading containers: start.
DEBU[0001] Loaded container 2a4f2f88d74dc70ded0b98c042c3a12afc7c8dc3431d4d3b8191c681f497136c
DEBU[0001] Loaded container 43e823ff51bad00c28c0b9d5bcfca1382423ea03d4d9e00778d9e9047257b657
DEBU[0001] Loaded container 4e97445c51758f15443bbb65a1e23c553166ff0c3ecb6c5c1ea6fe51a24b546a
DEBU[0001] Loaded container 5bac049d342b010171650b84ce11c16b6cd061060bedd7506400081a35e9f319
DEBU[0001] Loaded container 6932e010b365337376ea96ea8668f15d532d95238bc37f11336673c220862d7c
DEBU[0001] Loaded container 8332411fd6da161365e6753fbe8139175fb68c460f68cf5193a374b42b3ddd3d
DEBU[0001] Loaded container a32b5b04d239e8158c5db55475176568c4cb4da1b6ed8c70931a2b5ba7e7b026
DEBU[0001] Loaded container cbe74fc025ede33bce3b257a6ca2dc708c3543b3a185dd5a1df3b7132675c160
DEBU[0001] Loaded container d6db4e9fb84f922eed0670424cb3952d608aa79d27ca0348e67c91cb2e0e5357
DEBU[0001] Loaded container e9c49895f716032214008fd9f7094c095aef1875889316487551ab8a60cb8f4a
DEBU[0001] Option Experimental: false
DEBU[0001] Option DefaultDriver: bridge
DEBU[0001] Option DefaultNetwork: bridge
INFO[0001] Firewalld running: false
DEBU[0001] /sbin/iptables, [--wait --version]
DEBU[0001] /sbin/iptables, [--wait -t nat -D PREROUTING -m addrtype --dst-type LOCAL -j DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t nat -D OUTPUT -m addrtype --dst-type LOCAL ! --dst 127.0.0.0/8 -j DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t nat -D OUTPUT -m addrtype --dst-type LOCAL -j DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t nat -D PREROUTING]
DEBU[0001] /sbin/iptables, [--wait -t nat -D OUTPUT]
DEBU[0001] /sbin/iptables, [--wait -t nat -F DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t nat -X DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t filter -F DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t filter -X DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t filter -F DOCKER-ISOLATION]
DEBU[0001] /sbin/iptables, [--wait -t filter -X DOCKER-ISOLATION]
DEBU[0001] /sbin/iptables, [--wait -t nat -n -L DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t nat -N DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t filter -n -L DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t filter -n -L DOCKER-ISOLATION]
DEBU[0001] /sbin/iptables, [--wait -t filter -C DOCKER-ISOLATION -j RETURN]
DEBU[0001] /sbin/iptables, [--wait -I DOCKER-ISOLATION -j RETURN]
DEBU[0001] /sbin/iptables, [--wait -t nat -C POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE]
DEBU[0001] /sbin/iptables, [--wait -t nat -C DOCKER -i docker0 -j RETURN]
DEBU[0001] /sbin/iptables, [--wait -t nat -I DOCKER -i docker0 -j RETURN]
DEBU[0001] /sbin/iptables, [--wait -D FORWARD -i docker0 -o docker0 -j DROP]
DEBU[0001] /sbin/iptables, [--wait -t filter -C FORWARD -i docker0 -o docker0 -j ACCEPT]
DEBU[0001] /sbin/iptables, [--wait -t filter -C FORWARD -i docker0 ! -o docker0 -j ACCEPT]
DEBU[0001] /sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT]
DEBU[0001] /sbin/iptables, [--wait -t nat -C PREROUTING -m addrtype --dst-type LOCAL -j DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t nat -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t nat -C OUTPUT -m addrtype --dst-type LOCAL -j DOCKER ! --dst 127.0.0.0/8]
DEBU[0001] /sbin/iptables, [--wait -t nat -A OUTPUT -m addrtype --dst-type LOCAL -j DOCKER ! --dst 127.0.0.0/8]
DEBU[0001] /sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -j DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -j DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t filter -C FORWARD -j DOCKER-ISOLATION]
DEBU[0001] /sbin/iptables, [--wait -D FORWARD -j DOCKER-ISOLATION]
DEBU[0001] /sbin/iptables, [--wait -I FORWARD -j DOCKER-ISOLATION]
DEBU[0001] Network (bc712e4) restored
DEBU[0001] Allocating IPv4 pools for network bridge (bc712e4309a4900fcbb16ff116b3f5abbef7dae215a8debf9db185cf1d611ba3)
DEBU[0001] RequestPool(LocalDefault, 172.17.0.0/16, , map[], false)
DEBU[0001] RequestAddress(LocalDefault/172.17.0.0/16, 172.17.0.1, map[RequestAddressType:com.docker.network.gateway])
DEBU[0001] /sbin/iptables, [--wait -t nat -C POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE]
DEBU[0001] /sbin/iptables, [--wait -t nat -D POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE]
DEBU[0001] /sbin/iptables, [--wait -t nat -C DOCKER -i docker0 -j RETURN]
DEBU[0001] /sbin/iptables, [--wait -t nat -D DOCKER -i docker0 -j RETURN]
DEBU[0001] /sbin/iptables, [--wait -t filter -C FORWARD -i docker0 -o docker0 -j ACCEPT]
DEBU[0001] /sbin/iptables, [--wait -D FORWARD -i docker0 -o docker0 -j ACCEPT]
DEBU[0001] /sbin/iptables, [--wait -t filter -C FORWARD -i docker0 ! -o docker0 -j ACCEPT]
DEBU[0001] /sbin/iptables, [--wait -D FORWARD -i docker0 ! -o docker0 -j ACCEPT]
DEBU[0001] /sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT]
DEBU[0001] /sbin/iptables, [--wait -D FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT]
DEBU[0001] /sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -j DOCKER]
DEBU[0001] /sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -j DOCKER]
DEBU[0001] /sbin/iptables, [--wait -D FORWARD -o docker0 -j DOCKER]
DEBU[0001] releasing IPv4 pools from network bridge (bc712e4309a4900fcbb16ff116b3f5abbef7dae215a8debf9db185cf1d611ba3)
DEBU[0001] ReleaseAddress(LocalDefault/172.17.0.0/16, 172.17.0.1)
DEBU[0001] ReleasePool(LocalDefault/172.17.0.0/16)
INFO[0001] Default bridge (docker0) is assigned with an IP address 172.17.0.0/16. Daemon option --bip can be used to set a preferred IP address
DEBU[0001] Allocating IPv4 pools for network bridge (a07efb68a8b72d42c18c6d090bf5922bbe5c5b3f0eae87cf9b9d1297c192b8da)
DEBU[0001] RequestPool(LocalDefault, 172.17.0.0/16, , map[], false)
DEBU[0001] RequestAddress(LocalDefault/172.17.0.0/16, 172.17.0.1, map[RequestAddressType:com.docker.network.gateway])
DEBU[0002] /sbin/iptables, [--wait -t nat -C POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE]
DEBU[0002] /sbin/iptables, [--wait -t nat -I POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE]
DEBU[0002] /sbin/iptables, [--wait -t nat -C DOCKER -i docker0 -j RETURN]
DEBU[0002] /sbin/iptables, [--wait -t nat -I DOCKER -i docker0 -j RETURN]
DEBU[0002] /sbin/iptables, [--wait -D FORWARD -i docker0 -o docker0 -j DROP]
DEBU[0002] /sbin/iptables, [--wait -t filter -C FORWARD -i docker0 -o docker0 -j ACCEPT]
DEBU[0002] /sbin/iptables, [--wait -I FORWARD -i docker0 -o docker0 -j ACCEPT]
DEBU[0002] /sbin/iptables, [--wait -t filter -C FORWARD -i docker0 ! -o docker0 -j ACCEPT]
DEBU[0002] /sbin/iptables, [--wait -I FORWARD -i docker0 ! -o docker0 -j ACCEPT]
DEBU[0002] /sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT]
DEBU[0002] /sbin/iptables, [--wait -I FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT]
DEBU[0002] /sbin/iptables, [--wait -t nat -C PREROUTING -m addrtype --dst-type LOCAL -j DOCKER]
DEBU[0002] /sbin/iptables, [--wait -t nat -C PREROUTING -m addrtype --dst-type LOCAL -j DOCKER]
DEBU[0002] /sbin/iptables, [--wait -t nat -C OUTPUT -m addrtype --dst-type LOCAL -j DOCKER ! --dst 127.0.0.0/8]
DEBU[0002] /sbin/iptables, [--wait -t nat -C OUTPUT -m addrtype --dst-type LOCAL -j DOCKER ! --dst 127.0.0.0/8]
DEBU[0002] /sbin/iptables, [--wait -t filter -C FORWARD -o docker0 -j DOCKER]
DEBU[0002] /sbin/iptables, [--wait -I FORWARD -o docker0 -j DOCKER]
DEBU[0002] /sbin/iptables, [--wait -t filter -C FORWARD -j DOCKER-ISOLATION]
DEBU[0002] /sbin/iptables, [--wait -D FORWARD -j DOCKER-ISOLATION]
DEBU[0002] /sbin/iptables, [--wait -I FORWARD -j DOCKER-ISOLATION]
INFO[0002] Loading containers: done.
INFO[0002] Daemon has completed initialization
INFO[0002] Docker daemon                                 commit=60ccb22 graphdriver=devicemapper version=17.03.0-ce
DEBU[0002] Registering routers
DEBU[0002] Registering GET, /containers/{name:.*}/checkpoints
DEBU[0002] Registering POST, /containers/{name:.*}/checkpoints
DEBU[0002] Registering DELETE, /containers/{name}/checkpoints/{checkpoint}
DEBU[0002] Registering HEAD, /containers/{name:.*}/archive
DEBU[0002] Registering GET, /containers/json
DEBU[0002] Registering GET, /containers/{name:.*}/export
DEBU[0002] Registering GET, /containers/{name:.*}/changes
DEBU[0002] Registering GET, /containers/{name:.*}/json
DEBU[0002] Registering GET, /containers/{name:.*}/top
DEBU[0002] Registering GET, /containers/{name:.*}/logs
DEBU[0002] Registering GET, /containers/{name:.*}/stats
DEBU[0002] Registering GET, /containers/{name:.*}/attach/ws
DEBU[0002] Registering GET, /exec/{id:.*}/json
DEBU[0002] Registering GET, /containers/{name:.*}/archive
DEBU[0002] Registering POST, /containers/create
DEBU[0002] Registering POST, /containers/{name:.*}/kill
DEBU[0002] Registering POST, /containers/{name:.*}/pause
DEBU[0002] Registering POST, /containers/{name:.*}/unpause
DEBU[0002] Registering POST, /containers/{name:.*}/restart
DEBU[0002] Registering POST, /containers/{name:.*}/start
DEBU[0002] Registering POST, /containers/{name:.*}/stop
DEBU[0002] Registering POST, /containers/{name:.*}/wait
DEBU[0002] Registering POST, /containers/{name:.*}/resize
DEBU[0002] Registering POST, /containers/{name:.*}/attach
DEBU[0002] Registering POST, /containers/{name:.*}/copy
DEBU[0002] Registering POST, /containers/{name:.*}/exec
DEBU[0002] Registering POST, /exec/{name:.*}/start
DEBU[0002] Registering POST, /exec/{name:.*}/resize
DEBU[0002] Registering POST, /containers/{name:.*}/rename
DEBU[0002] Registering POST, /containers/{name:.*}/update
DEBU[0002] Registering POST, /containers/prune
DEBU[0002] Registering PUT, /containers/{name:.*}/archive
DEBU[0002] Registering DELETE, /containers/{name:.*}
DEBU[0002] Registering GET, /images/json
DEBU[0002] Registering GET, /images/search
DEBU[0002] Registering GET, /images/get
DEBU[0002] Registering GET, /images/{name:.*}/get
DEBU[0002] Registering GET, /images/{name:.*}/history
DEBU[0002] Registering GET, /images/{name:.*}/json
DEBU[0002] Registering POST, /commit
DEBU[0002] Registering POST, /images/load
DEBU[0002] Registering POST, /images/create
DEBU[0002] Registering POST, /images/{name:.*}/push
DEBU[0002] Registering POST, /images/{name:.*}/tag
DEBU[0002] Registering POST, /images/prune
DEBU[0002] Registering DELETE, /images/{name:.*}
DEBU[0002] Registering OPTIONS, /{anyroute:.*}
DEBU[0002] Registering GET, /_ping
DEBU[0002] Registering GET, /events
DEBU[0002] Registering GET, /info
DEBU[0002] Registering GET, /version
DEBU[0002] Registering GET, /system/df
DEBU[0002] Registering POST, /auth
DEBU[0002] Registering GET, /volumes
DEBU[0002] Registering GET, /volumes/{name:.*}
DEBU[0002] Registering POST, /volumes/create
DEBU[0002] Registering POST, /volumes/prune
DEBU[0002] Registering DELETE, /volumes/{name:.*}
DEBU[0002] Registering POST, /build
DEBU[0002] Registering POST, /swarm/init
DEBU[0002] Registering POST, /swarm/join
DEBU[0002] Registering POST, /swarm/leave
DEBU[0002] Registering GET, /swarm
DEBU[0002] Registering GET, /swarm/unlockkey
DEBU[0002] Registering POST, /swarm/update
DEBU[0002] Registering POST, /swarm/unlock
DEBU[0002] Registering GET, /services
DEBU[0002] Registering GET, /services/{id}
DEBU[0002] Registering POST, /services/create
DEBU[0002] Registering POST, /services/{id}/update
DEBU[0002] Registering DELETE, /services/{id}
DEBU[0002] Registering GET, /services/{id}/logs
DEBU[0002] Registering GET, /nodes
DEBU[0002] Registering GET, /nodes/{id}
DEBU[0002] Registering DELETE, /nodes/{id}
DEBU[0002] Registering POST, /nodes/{id}/update
DEBU[0002] Registering GET, /tasks
DEBU[0002] Registering GET, /tasks/{id}
DEBU[0002] Registering GET, /secrets
DEBU[0002] Registering POST, /secrets/create
DEBU[0002] Registering DELETE, /secrets/{id}
DEBU[0002] Registering GET, /secrets/{id}
DEBU[0002] Registering POST, /secrets/{id}/update
DEBU[0002] Registering GET, /plugins
DEBU[0002] Registering GET, /plugins/{name:.*}/json
DEBU[0002] Registering GET, /plugins/privileges
DEBU[0002] Registering DELETE, /plugins/{name:.*}
DEBU[0002] Registering POST, /plugins/{name:.*}/enable
DEBU[0002] Registering POST, /plugins/{name:.*}/disable
DEBU[0002] Registering POST, /plugins/pull
DEBU[0002] Registering POST, /plugins/{name:.*}/push
DEBU[0002] Registering POST, /plugins/{name:.*}/upgrade
DEBU[0002] Registering POST, /plugins/{name:.*}/set
DEBU[0002] Registering POST, /plugins/create
DEBU[0002] Registering GET, /networks
DEBU[0002] Registering GET, /networks/
DEBU[0002] Registering GET, /networks/{id:.+}
DEBU[0002] Registering POST, /networks/create
DEBU[0002] Registering POST, /networks/{id:.*}/connect
DEBU[0002] Registering POST, /networks/{id:.*}/disconnect
DEBU[0002] Registering POST, /networks/prune
DEBU[0002] Registering DELETE, /networks/{id:.*}
INFO[0002] API listen on /var/run/docker.sock
```

**Debug log output of 'docker run hello-world'**

```
DEBU[0008] Calling GET /_ping
DEBU[0008] Calling POST /v1.26/containers/create
DEBU[0008] form data: {"AttachStderr":true,"AttachStdin":false,"AttachStdout":true,"Cmd":null,"Domainname":"","Entrypoint":null,"Env":[],"HostConfig":{"AutoRemove":false,"Binds":null,"BlkioDeviceReadBps":null,"BlkioDeviceReadIOps":null,"BlkioDeviceWriteBps":null,"BlkioDeviceWriteIOps":null,"BlkioWeight":0,"BlkioWeightDevice":null,"CapAdd":null,"CapDrop":null,"Cgroup":"","CgroupParent":"","ConsoleSize":[0,0],"ContainerIDFile":"","CpuCount":0,"CpuPercent":0,"CpuPeriod":0,"CpuQuota":0,"CpuRealtimePeriod":0,"CpuRealtimeRuntime":0,"CpuShares":0,"CpusetCpus":"","CpusetMems":"","Devices":[],"DiskQuota":0,"Dns":[],"DnsOptions":[],"DnsSearch":[],"ExtraHosts":null,"GroupAdd":null,"IOMaximumBandwidth":0,"IOMaximumIOps":0,"IpcMode":"","Isolation":"","KernelMemory":0,"Links":null,"LogConfig":{"Config":{},"Type":""},"Memory":0,"MemoryReservation":0,"MemorySwap":0,"MemorySwappiness":-1,"NanoCpus":0,"NetworkMode":"default","OomKillDisable":false,"OomScoreAdj":0,"PidMode":"","PidsLimit":0,"PortBindings":{},"Privileged":false,"PublishAllPorts":false,"ReadonlyRootfs":false,"RestartPolicy":{"MaximumRetryCount":0,"Name":"no"},"SecurityOpt":null,"ShmSize":0,"UTSMode":"","Ulimits":null,"UsernsMode":"","VolumeDriver":"","VolumesFrom":null},"Hostname":"","Image":"hello-world","Labels":{},"NetworkingConfig":{"EndpointsConfig":{}},"OnBuild":null,"OpenStdin":false,"StdinOnce":false,"Tty":false,"User":"","Volumes":{},"WorkingDir":""}
DEBU[0008] devmapper: AddDevice START(hash=e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init basehash=c2eca2c3d1d01bb05528edc5269f9603c91c8427222fea9d5b60371fee62c13b)
DEBU[0008] devmapper: Failed to read /var/virtual/docker/10000.10000/devicemapper/metadata/e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init with err: open /var/virtual/docker/10000.10000/devicemapper/metadata/e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init: no such file or directory
DEBU[0008] devmapper: registerDevice(23, e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init)
DEBU[0008] devmapper: AddDevice END(hash=e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init basehash=c2eca2c3d1d01bb05528edc5269f9603c91c8427222fea9d5b60371fee62c13b)
DEBU[0008] devmapper: activateDeviceIfNeeded(e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init)
DEBU[0008] devmapper: UnmountDevice START(hash=e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init)
DEBU[0008] devmapper: Unmount(/var/virtual/docker/10000.10000/devicemapper/mnt/e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init)
DEBU[0008] devmapper: Unmount done
DEBU[0008] devmapper: deactivateDevice START(e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init)
DEBU[0008] devmapper: removeDevice START(docker-253:12-2108032-e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init)
DEBU[0008] devmapper: removeDevice END(docker-253:12-2108032-e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init)
DEBU[0008] devmapper: deactivateDevice END(e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init)
DEBU[0008] devmapper: UnmountDevice END(hash=e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init)
DEBU[0008] devmapper: AddDevice START(hash=e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233 basehash=e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init)
DEBU[0008] devmapper: Failed to read /var/virtual/docker/10000.10000/devicemapper/metadata/e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233 with err: open /var/virtual/docker/10000.10000/devicemapper/metadata/e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233: no such file or directory
DEBU[0008] devmapper: registerDevice(24, e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0008] devmapper: AddDevice END(hash=e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233 basehash=e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233-init)
DEBU[0009] devmapper: activateDeviceIfNeeded(e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0009] container mounted via layerStore: /var/virtual/docker/10000.10000/devicemapper/mnt/e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233/rootfs
DEBU[0009] devmapper: UnmountDevice START(hash=e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0009] devmapper: Unmount(/var/virtual/docker/10000.10000/devicemapper/mnt/e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0009] devmapper: Unmount done
DEBU[0009] devmapper: deactivateDevice START(e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0009] devmapper: removeDevice START(docker-253:12-2108032-e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0009] devmapper: removeDevice END(docker-253:12-2108032-e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0009] devmapper: deactivateDevice END(e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0009] devmapper: UnmountDevice END(hash=e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0009] Calling POST /v1.26/containers/68f485cc63714a188057a76d38524b6e2806c4df642827021f7b061343c9bd42/attach?stderr=1&stdout=1&stream=1
DEBU[0009] attach: stdout: begin
DEBU[0009] attach: stderr: begin
DEBU[0009] Calling GET /v1.26/events?filters=%7B%22container%22%3A%7B%2268f485cc63714a188057a76d38524b6e2806c4df642827021f7b061343c9bd42%22%3Atrue%7D%2C%22type%22%3A%7B%22container%22%3Atrue%7D%7D
DEBU[0009] Calling POST /v1.26/containers/68f485cc63714a188057a76d38524b6e2806c4df642827021f7b061343c9bd42/start
DEBU[0009] devmapper: activateDeviceIfNeeded(e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0009] container mounted via layerStore: /var/virtual/docker/10000.10000/devicemapper/mnt/e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233/rootfs
DEBU[0009] Assigning addresses for endpoint hardcore_hawking's interface on network bridge
DEBU[0009] RequestAddress(LocalDefault/172.17.0.0/16, <nil>, map[])
DEBU[0009] Assigning addresses for endpoint hardcore_hawking's interface on network bridge
INFO[0009] No non-localhost DNS nameservers are left in resolv.conf. Using default external servers: [nameserver 8.8.8.8 nameserver 8.8.4.4]
INFO[0009] IPv6 enabled; Adding default IPv6 external servers: [nameserver 2001:4860:4860::8888 nameserver 2001:4860:4860::8844]
DEBU[0009] Programming external connectivity on endpoint hardcore_hawking (da1229a48eb767ae02bb07c9a3f7c66adc8177d3edcb5cf875fe37b45fc10999)
ERRO[0010] containerd: start container                   error=oci runtime error: container_linux.go:247: starting container process caused "process_linux.go:359: container init caused \"rootfs_linux.go:54: mounting \\\"proc\\\" to rootfs \\\"/var/virtual/docker/10000.10000/devicemapper/mnt/e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233/rootfs\\\" at \\\"/proc\\\" caused \\\"operation not permitted\\\"\""
 id=68f485cc63714a188057a76d38524b6e2806c4df642827021f7b061343c9bd42
ERRO[0010] Create container failed with error: oci runtime error: container_linux.go:247: starting container process caused "process_linux.go:359: container init caused \"rootfs_linux.go:54: mounting \\\"proc\\\" to rootfs \\\"/var/virtual/docker/10000.10000/devicemapper/mnt/e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233/rootfs\\\" at \\\"/proc\\\" caused \\\"operation not permitted\\\"\""

DEBU[0010] attach: stdout: end
DEBU[0010] attach: stderr: end
DEBU[0010] Revoking external connectivity on endpoint hardcore_hawking (da1229a48eb767ae02bb07c9a3f7c66adc8177d3edcb5cf875fe37b45fc10999)
DEBU[0010] Releasing addresses for endpoint hardcore_hawking's interface on network bridge
DEBU[0010] ReleaseAddress(LocalDefault/172.17.0.0/16, 172.17.0.2)
DEBU[0010] devmapper: UnmountDevice START(hash=e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0010] devmapper: Unmount(/var/virtual/docker/10000.10000/devicemapper/mnt/e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0010] devmapper: Unmount done
DEBU[0010] devmapper: deactivateDevice START(e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0010] devmapper: removeDevice START(docker-253:12-2108032-e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0010] devmapper: removeDevice END(docker-253:12-2108032-e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0010] devmapper: deactivateDevice END(e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
DEBU[0010] devmapper: UnmountDevice END(hash=e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233)
ERRO[0010] Handler for POST /v1.26/containers/68f485cc63714a188057a76d38524b6e2806c4df642827021f7b061343c9bd42/start returned error: oci runtime error: container_linux.go:247: starting container process caused "process_linux.go:359: container init caused \"rootfs_linux.go:54: mounting \\\"proc\\\" to rootfs \\\"/var/virtual/docker/10000.10000/devicemapper/mnt/e42fb9f0d9e65fb8791a43aad42d2b9c0b88ccd99cbadf0959f741db5cdff233/rootfs\\\" at \\\"/proc\\\" caused \\\"operation not permitted\\\"\""

DEBU[0010] Client context cancelled, stop sending events
```

**Output of 'docker version'**
```
Client:
 Version:      17.03.0-ce
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   60ccb22
 Built:        Thu Feb 23 10:53:29 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.0-ce
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   60ccb22
 Built:        Thu Feb 23 10:53:29 2017
 OS/Arch:      linux/amd64
 Experimental: false

```

**Output of 'docker info'**
```
Containers: 11
 Running: 0
 Paused: 0
 Stopped: 11
Images: 1
Server Version: 17.03.0-ce
Storage Driver: devicemapper
 Pool Name: storage-docker-tpool
 Pool Blocksize: 524.3 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: ext4
 Data file:
 Metadata file:
 Data Space Used: 356.5 MB
 Data Space Total: 214.7 GB
 Data Space Available: 214.4 GB
 Metadata Space Used: 675.8 kB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.147 GB
 Thin Pool Minimum Free Space: 21.47 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.90 (2014-09-01)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 977c511eda0925a723debdc94d09459af49d082a
runc version: a01dafd48bc1c7cc12bdb01206f9fea7dd6feb70
init version: 949e6fa
Security Options:
 userns
Kernel Version: 3.16.0-4-amd64
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.41 GiB
Name: REDACTED
ID: TL2R:7UQW:ZRXP:4Y6J:7L72:D4DV:TXCN:XUG5:6FWO:RMIS:ZKCS:CWOA
Docker Root Dir: /var/virtual/docker/10000.10000
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No kernel memory limit support
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by ziermmar** [comment]
Still happening in version 17.04

**Comment by estesp** [comment]
I installed a Debian Jessie VM; I have 17.03.1-ce installed, and turned on user namespaces in the daemon. I tried both aufs and devicemapper as backends, but neither one generated the `/proc` mounting error.

I've seen what I believe was a similar error a long time ago--the context was someone trying out user namespace support in the engine, but had an Ubuntu system where they attempted a more recent Ubuntu release kernel with a unique (non-matching release) userspace. IIRC, manually mounting `/proc` on the host before starting Docker was a workaround for that weird mismatched version situation, but I'm not sure if that has any relation to what you are seeing.

**Comment by ziermmar** [comment]
```
# mount | grep proc                                                                                                                                           
proc on /proc type proc (rw,relatime,hidepid=2)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=22,pgrp=1,timeout=300,minproto=5,maxproto=5,direct)
nfsd on /proc/fs/nfsd type nfsd (rw,relatime)
binfmt_misc on /proc/sys/fs/binfmt_misc type binfmt_misc (rw,relatime)
```

Well let's see. Those are my mounts. I could'nt remount proc without touching those depending on /proc, so I unmounted them first and voila, docker ran again. I don't know yet which one was preventing docker from running, but thanks, @estesp your hint helped me out!

**Comment by thaJeztah** [comment]
Good to hear; looks like this issue was resolved, so let me go ahead and close 👍
